### PR TITLE
add circle linked list in C #3780

### PR DESCRIPTION
**Pull request by GP1999**
**Fixes issue:**
add circle linked list in C #3780 

**Changes:**
 I Have implemented circular linked list

### CODE REVIEW

File: code/data_structures/src/list/circular_linked_list/circular_linked_list.c [hunk 1]
```diff
@@ -0,0 +1,178 @@
+
```
Comment (arnavb) [inline code/data_structures/src/list/circular_linked_list/circular_linked_list.c]: Please follow our C style guide.

File: code/data_structures/src/list/circular_linked_list/circular_linked_list.c [hunk 2]
```diff
@@ -0,0 +1,178 @@
+
+#include<stdio.h>
+#include<stdlib.h>
+
+void insert_at_end();
+void insert_at_front();
+void delete_end();
+void delete_front();
+void display();
+struct Node *root=NULL,*tail=NULL;
```
Comment (arnavb) [inline code/data_structures/src/list/circular_linked_list/circular_linked_list.c]: Avoid global variables.

File: code/data_structures/src/list/circular_linked_list/circular_linked_list.c [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+#include<stdio.h>
+#include<stdlib.h>
+
+void insert_at_end();
+void insert_at_front();
+void delete_end();
+void delete_front();
+void display();
+struct Node *root=NULL,*tail=NULL;
+struct Node
```
Comment (arnavb) [inline code/data_structures/src/list/circular_linked_list/circular_linked_list.c]: Use a `typedef`.

**Comment by GP1999** [comment]
i have update changes you have suggested .please review it.

File: code/data_structures/src/list/circular_linked_list/circular_linked_list.c [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+#include<stdlib.h>
+typedef struct Node node;
+void display(node *tail);
+node *insert_at_end(node *tail);
+node *insert_at_front(node *tail);
+node *delete_at_begin(node *tail);
+node *delete_at_end(node *tail);
+
+struct Node{
+int Data;
```
Comment (arnavb) [inline code/data_structures/src/list/circular_linked_list/circular_linked_list.c]: Fix indentation here.

Comment (GP1999) [reply code/data_structures/src/list/circular_linked_list/circular_linked_list.c]: sorry,But where.In structure or in declaration section of functions.

Comment (GP1999) [reply code/data_structures/src/list/circular_linked_list/circular_linked_list.c]: i have updated now.Please review it.

**Comment by arnavb** [comment]
djldk 2

**Comment by arnavb** [comment]
@OpenGenus/cosmos-maintainers Please review!

**Comment by GP1999** [comment]
why merging is block?

> 
> 
> @OpenGenus/cosmos-maintainers Please review!

**Comment by GP1999** [comment]
why my pull request is not merged?Do i need more reviewer ?please guide me because i am new to open source.

**Comment by arnavb** [comment]
@GP1999 Yes. It says that 6 approving reviews are required.

**Comment by GP1999** [comment]
So how can i request to reviewers who has write access ?

**Comment by arnavb** [comment]
@GP1999 You need reviews from @OpenGenus/cosmos-maintainers.

**Comment by arnavb** [comment]
It turns out that the base branch you were merging to was incorrect; it should be master, not r0.1.

djldk 2

**Pull request merged**